llvm/lib/Target/EVM/EVMOptimizedCodeTransform.cpp

```diff
@@ -248,39 +253,92 @@ void EVMOptimizedCodeTransform::createStackLayout(Stack TargetStack) {
   assert(Assembly.getStackHeight() == static_cast<int>(CurrentStack.size()));
 }
 
-void EVMOptimizedCodeTransform::operator()(CFG::BasicBlock const &Block) {
+void EVMOptimizedCodeTransform::createOperationEntryLayout(
+    const CFG::Operation &Op) {
+  // Create required layout for entering the Operation.
+  // Check if we can choose cheaper stack shuffling if the Operation is an
+  // instruction with commutable arguments.
+  if (const auto *Inst = std::get_if<CFG::BuiltinCall>(&Op.Operation);
+      Inst && Inst->IsCommutable) {
+    // Get the stack layout before the instruction.
+    const Stack &DefaultTargetStack = Layout.operationEntryLayout.at(&Op);
+    size_t DefaultCost =
+        EvaluateStackTransform(CurrentStack, DefaultTargetStack);
+
+    // Commutable operands always take top two stack slots.
+    const unsigned OpIdx1 = 0, OpIdx2 = 1;
+    // Swap the commutable stack items and measure the stack shuffling cost
+    // again.
+    assert(DefaultTargetStack.size() > OpIdx1 &&
```

Comment:
Can it be just `DefaultTargetStack.size() > 1`?

Reply:
Thank you, fixed.